libsolidity/ast/Types.cpp

```diff
@@ -2422,12 +2422,18 @@ u256 TupleType::storageSize() const
 	solAssert(false, "Storage size of non-storable tuple type requested.");
 }
 
-unsigned TupleType::sizeOnStack() const
+vector<string> TupleType::makeStackSlotNames() const
 {
-	unsigned size = 0;
+	vector<string> slotNames;
+	unsigned i = 1;
 	for (auto const& t: components())
-		size += t ? t->sizeOnStack() : 0;
-	return size;
+	{
+		if (t)
+			for (auto const& componentName: t->stackSlotNames())
+				slotNames.emplace_back(std::to_string(i) + (componentName.empty() ? "" : "_" + componentName));
```

Comment:
I think it might be safer to also prepend `tuple_` or `component_`. Otherwise we might get into issues where the naming scheme is not prefix free anymore.

Reply:
I was also wondering, whether we should pick a different delimiter instead of ``_`` for this... like ``$`` or ``$$``...